tests/infer/mcmc/test_mh.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import, division, print_function
+
+import pytest
+import torch
+from torch.autograd import Variable
+
+import pyro
+import pyro.infer
+from pyro.distributions import Normal
+from tests.common import TestCase
+from pyro.infer.abstract_infer import Marginal
+from pyro.infer.mcmc.mh import MH, NormalProposal
+from pyro.infer.mcmc.mcmc import MCMC
+from pyro.util import ng_ones, ng_zeros
+
+
+class NormalNormalSamplingTestCase(TestCase):
+
+    def setUp(self):
+
+        pyro.clear_param_store()
+
+        def model():
+            mu = pyro.sample("mu", Normal(Variable(torch.zeros(1)),
+                                          Variable(torch.ones(1))))
+            xd = Normal(mu, Variable(torch.ones(1)), batch_size=50)
+            pyro.observe("xs", xd, self.data)
+            return mu
+
+        # data
+        self.data = Variable(torch.zeros(50, 1))
+        self.mu_mean = Variable(torch.zeros(1))
+        self.mu_stddev = torch.sqrt(Variable(torch.ones(1)) / 51.0)
+
+        # model and guide
+        self.model = model
+
+
+class MHTest(NormalNormalSamplingTestCase):
+
+    @pytest.mark.init(rng_seed=0)
+    def test_single_site_mh(self):
+
+        # basic mcmc run with normal normal
+        mcmc_run = MCMC(MH(self.model, NormalProposal(mu=ng_zeros(1), sigma=ng_ones(1), tune_frequency=100)),
+                        num_samples=1000, warmup_steps=150)
+
+        marginal = Marginal(mcmc_run)
+        posterior_samples = [marginal() for i in range(1000)]
+
+        posterior_mean = torch.mean(torch.cat(posterior_samples))
+        posterior_stddev = torch.std(torch.cat(posterior_samples), 0)
+
+        self.assertEqual(0, torch.norm(posterior_mean - self.mu_mean).data[0],
```

Comment:
Prefer the nice `assert_equal` helper from test/common.py:
```py
from common import assert_equal
assert_equal(posterior_mean, self.mu_mean, prec=0.03)
```
Also nit, a nice convention I like to follow in testing is to name these `self.expected_mean` and then compute `actual_mean = torch.mean(...)` and check `assert_equal(actual_mean, expected_mean, prec=0.03)`. The advantage of this naming convention is that when a maintainer breaks some upstream piece of code during refactoring, it's clear which value is known truth (`expected`) and which is the value under test (`actual`).

Reply:
Happy to change this, it seemed a bit odd to me as well. However, I remain a bit confused mainly because this is copied directly from an [existing pyro test](https://github.com/uber/pyro/blob/dev/tests/infer/test_sampling.py#L111). 

In other words, I was under the assumption that existing tests were a model for new tests. Is this not the case?

Reply:
Some of Pyro's tests still use older unittest assertions, but newer `assert` statements are now preferred. If you're ever modifying a test file in older style, feel free to clean up and convert everything to newer style.

Reply:
Also, `test_sampling.py` is kind of a mess, at least statistically.  We should rewrite it now that we're adding more sampling algorithms.